src/checkerTypes.ml

```diff
@@ -56,4 +56,10 @@ type view_current_liquidation_auction_minimum_bid_result =
   }
 [@@deriving show]
 
+type view_current_liquidation_auction_remaining_duration_result =
+  { blocks: Ligo.int
+  ; seconds: Ligo.int
+  }
+[@@deriving show]
```

Comment:
I see `view_current_liquidation_auction_minimum_bid_result` and `view_current_liquidation_auction_remaining_duration_result` and all the duplication in the views and I am wondering if it would be better to just have a big `current_liquidation_auction_data` type and one massive view returning everything about it (something like `view_current_auction_data`) :thinking: @dorranh what do you think? If I remember correctly  @utdemir had suggested something along these lines at some point as well.

Reply:
Hmm that's a good point and could make more sense for these current auction views specifically. One thing we might want to watch out for is combining more efficient views with ones which have to do things like traversing the queue since users might want to do things like poll easy-to-lookup data such as the current bid. This is not an issue for this set of views though since we are only performing lookups and some arithmetic. I'll take a look at merging them.

Reply:
Yeah, good point :+1: I'd say we can at least group the following "cheap" ones regarding the current auction:

* id
* amount of tez being auctioned off
* currently winning bid
* minimum bid to place
* blocks remaining until closing
* seconds remaining until closing

If we group them together then I think your initial approach of using `option` is better, since we don't want to fail to give any result if for example nobody has bid yet (i.e., there is no "currently winning bid"). So, you could get the current auction via `liquidation_auction_get_current_auction` which could fail if there is no current auction, but after that most of the fields in the returned data structure should probably be `option`s.

For the expensive entrypoints I think we could ask them individually indeed